src/hypertable_restrict_info.c

```diff
@@ -71,68 +73,108 @@ dimension_restrict_info_create(Dimension *d)
 }
 
 static bool
-dimension_restrict_info_open_add(DimensionRestrictInfoOpen *dri, StrategyNumber strategy, Const *c)
+dimension_restrict_info_open_add(DimensionRestrictInfoOpen *dri, StrategyNumber strategy, List *values, Oid type)
```

Comment:
This does not seem correct to me. It again handles the OR case the same as the AND case and this isn't right. `WHERE time < '2001-01-01' AND time < '2000-01-01'` seems like a different case than `WHERE time < ANY ('2001-01-01', '2000-01-01')`. But this code will treat it equivalently?

Reply:
Related question how does this code handle differences between ANY and ALL

Reply:
Actually this PR is not addressing open dimensions like time. I think we went through it already and decided not to support IN/ANY for time dimension. IN/ANY/ALL are all handled in the same way.

Reply:
We filter out ANY/ALL/IN here (if we get `-1` when trying to convert Datum to int64) https://github.com/timescale/timescaledb/pull/784/commits/f14c8414c21678aa1512d1e6682f8068b5fa66b9#diff-273cc5022b4d73507f7284cbee02026bR98